drivers: flash: common: drivers.flash.common.disable_spi_nor fails

**Describe the bug**
The test `tests/drivers/flash/common/drivers.flash.common.disable_spi_nor` fails.

Observed for:
- `nrf54l15dk/nrf54l05/cpuapp`
- `nrf54l15dk/nrf54l10/cpuapp`
- `nrf54l15dk/nrf54l15/cpuapp`

**To Reproduce**
Steps to reproduce the behavior:
1. go to your zephyr dir
2. call `./scripts/twister -s drivers.flash.common.disable_spi_nor -p nrf54l15dk/nrf54l15/cpuapp --device-testing --device-serial /dev/ttyACM1 -v --inline-logs --west-flash="--erase"`
3. See console output with error

**Expected behavior**
Valid console output

**Impact**
Not clear

**Logs and console output**
```
*** Booting Zephyr OS build v4.1.0-3766-gb2c6d960cbd1 ***
Running TESTSUITE flash_driver
===================================================================
START - test_flash_copy
Test will run on device rram-controller@5004b000
No devices with erase requirement present

    Assertion failed at zephyr/tests/drivers/flash/common/src/main.c:403: test_flash_copy_inner: flash_flatten(flash_dev, page_info.start_offset + page_info.size, page_info.size) is non-zero

 FAIL - test_flash_copy in 0.246 seconds
===================================================================
START - test_flash_erase
Test will run on device rram-controller@5004b000
No devices with erase requirement present
 PASS - test_flash_erase in 0.155 seconds
===================================================================
START - test_flash_fill
Test will run on device rram-controller@5004b000
No devices with erase requirement present
 PASS - test_flash_fill in 0.423 seconds
===================================================================
START - test_flash_flatten
Test will run on device rram-controller@5004b000
No devices with erase requirement present
 PASS - test_flash_flatten in 0.351 seconds
===================================================================
START - test_flash_page_layout
Test will run on device rram-controller@5004b000
No devices with erase requirement present
start_offset=0x15c000size=4096index=348
page_count=357
 PASS - test_flash_page_layout in 0.014 seconds
===================================================================
START - test_get_size
Test will run on device rram-controller@5004b000
No devices with erase requirement present
 SKIP - test_get_size in 0.009 seconds
===================================================================
START - test_read_unaligned_address
Test will run on device rram-controller@5004b000
No devices with erase requirement present
 PASS - test_read_unaligned_address in 0.013 seconds
===================================================================
TESTSUITE flash_driver failed.

------ TESTSUITE SUMMARY START ------

SUITE FAIL -  83.33% [flash_driver]: pass = 5, fail = 1, skip = 1, total = 7 duration = 1.211 seconds
 - FAIL - [flash_driver.test_flash_copy] duration = 0.246 seconds
 - PASS - [flash_driver.test_flash_erase] duration = 0.155 seconds
 - PASS - [flash_driver.test_flash_fill] duration = 0.423 seconds
 - PASS - [flash_driver.test_flash_flatten] duration = 0.351 seconds
 - PASS - [flash_driver.test_flash_page_layout] duration = 0.014 seconds
 - SKIP - [flash_driver.test_get_size] duration = 0.009 seconds
 - PASS - [flash_driver.test_read_unaligned_address] duration = 0.013 seconds

------ TESTSUITE SUMMARY END ------

===================================================================
PROJECT EXECUTION FAILED
```

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04.1 LTS
 - Toolchain Zephyr SDK 0.17.0
 - Commit SHA or Version used: `b2c6d960cbd1de8258d18acfcf947ca13231e84d`

**Additional information:**
Probably introducedy by commit d8be41c00a1ca2dcdc97031436dc851a6f5137b9

Seen on fk7b0m1_vbt6 board when building tests/drivers/flash/common -T drivers.flash.common.disable_spi_nor
```
in function `flash_erase':
~/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/syscalls/flash.h:90: undefined reference to `__device_dts_ord_132'
~/zephyr-sdk-0.16.8/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: app/libapp.a(main.c.obj): in function `z_impl_flash_read':
~/zephyrproject/zephyr/include/zephyr/drivers/flash.h:255: undefined reference to `__device_dts_ord_132'
~/zephyr-sdk-0.16.8/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: app/libapp.a(main.c.obj): in function `z_impl_flash_get_size':
~/zephyrproject/zephyr/include/zephyr/drivers/flash.h:354: undefined reference to `__device_dts_ord_132'
~/zephyr-sdk-0.16.8/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: app/libapp.a(main.c.obj): in function `flash_driver_before':
~/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:124: undefined reference to `__device_dts_ord_132'
~/zephyr-sdk-0.16.8/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: app/libapp.a(main.c.obj): in function `_flash_driver_test_flash_page_layout_wrapper':
~/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:344: undefined reference to `__device_dts_ord_132'
```

--- a/drivers/flash/flash_util.c
+++ b/drivers/flash/flash_util.c
@@ -59,7 +59,7 @@ int z_impl_flash_fill(const struct device *dev, uint8_t val, off_t offset,
 
 		rc = api->write(dev, offset + stored, filler, chunk);
 		if (rc < 0) {
-			LOG_DBG("Fill to dev %p failed at offset 0x%zx\n",
+			LOG_ERR("Fill to dev %p failed at offset 0x%zx\n",
 				dev, (size_t)offset + stored);
 			break;
 		}

--- a/tests/drivers/flash/common/src/main.c
+++ b/tests/drivers/flash/common/src/main.c
@@ -79,7 +79,8 @@ static void flash_driver_before(void *arg)
 		TC_PRINT("No devices with erase requirement present\n");
 		erase_value = 0x55;
 		page_info.start_offset = TEST_AREA_OFFSET;
-		page_info.size = TEST_AREA_MAX - TEST_AREA_OFFSET;
+		/* test_flash_copy uses 2 pages, so split the test area */
+		page_info.size = (TEST_AREA_MAX - TEST_AREA_OFFSET) / 2;
 	}
 
 
@@ -102,6 +103,10 @@ static void flash_driver_before(void *arg)
 	/* Check if tested region fits in flash */
 	zassert_true((TEST_AREA_OFFSET + EXPECTED_SIZE) <= TEST_AREA_MAX,
 		     "Test area exceeds flash size");
+
+	/* Check if test region is suitable for test_flash_copy */
+	zassert_true((TEST_AREA_OFFSET + 2 * page_info.size) <= TEST_AREA_MAX,
+		     "test_flash_copy needs 2 flash pages");
 
 	/* Check if flash is cleared */
 	if (IS_ENABLED(CONFIG_FLASH_HAS_EXPLICIT_ERASE) && ebw_required) {

